Code/Framework/AzToolsFramework/AzToolsFramework/UI/LegacyFramework/Core/EditorFrameworkApplication.h

```diff
@@ -51,6 +51,7 @@ namespace LegacyFramework
     {
         /// Create application, if systemEntityFileName is NULL, we will create with default settings.
     public:
+        AZ_CLASS_ALLOCATOR(Application, AZ::SystemAllocator, 0);
```

Comment:
Question: How does the allocator resolve situations where the class name is duplicated in another namespace?

Reply:
The `AZ_CLASS_ALLOCATOR` macro expands to a bunch of overloads of `operator new()` and `operator delete()`. In this case, it would expand to:
```cpp
namespace LegacyFramework
{
// ... snip ...
    class Application
        : public AZ::ComponentApplication
        , protected FrameworkApplicationMessages::Handler
        , protected CoreMessageBus::Handler
    {
    public:
        //AZ_CLASS_ALLOCATOR(Application, AZ::SystemAllocator, 0);
        // the above macro will add, for example:
        AZ_FORCE_INLINE void* operator new(std::size_t size) {
            AZ_Assert(size == sizeof(Application), "Size mismatch! Did you forget to declare the macro in derived class? Size: %d sizeof(%s): %d", size, "Application", sizeof(Application));
            AZ_Warning(0, true/*false*/, "Make sure you use aznew, offers better tracking! (%s)", "Application");
            return AZ::AllocatorInstance< AZ::SystemAllocator >::Get().allocate(size, AZStd::alignment_of< Application >::value);
        }
```

At this point, the normal C++ name lookup rules apply. `AZStd::alignment_of< Application >::value` will look in the current namespace (`LegacyFramework` in this case), but not in others.

Reply:
Thanks for the education